modal/_container_exec.py

```diff
@@ -44,7 +37,6 @@ async def container_exec(
                 task_id=task_id,
                 command=command,
                 pty_info=get_pty_info(shell=True) if pty else None,
-                terminate_container_on_exit=terminate_container_on_exit,
```

Comment:
We can't remove `terminate_container_on_exit` from the schema yet. Here is the order of roll-out IMO:

- remove from code so clients don't sends terminate_container_on_exit as `True` anymore
- wait until everyone uses the new client.
- remove the usage of `terminate_on_exit` from `modal-runtime`
- remove the schema in `client`